### PR TITLE
New Semantic Memory Storage based on Azure Cognitive Search with custom embeddings

### PR DESCRIPTION
**Pull request by dluc**
* Add new Memory Storage based on Azure Cognitive Search with custom embeddings
* Delete previous Azure Search memory component based on old semantic search/Lucene

### CODE REVIEW

**Review by glahaye — APPROVED** [review]

**Comment by 1983Thai** [comment]
https://github.com/microsoft/semantic-kernel/pull/1248#pullrequestreview-1446913040

**Comment by dmytrostruk** [comment]
Status: waiting for NuGet to be available.

File: dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemory.cs [hunk 1]
```diff
@@ -1,355 +0,0 @@
...
-using Azure.Search.Documents.Models;
-using Microsoft.SemanticKernel.Memory;
-
-namespace Microsoft.SemanticKernel.Connectors.Memory.AzureCognitiveSearch;
-
-/// <summary>
-/// Semantic Memory implementation using Azure Cognitive Search.
-/// For more information about Azure Cognitive Search see https://learn.microsoft.com/azure/search/search-what-is-azure-search
-/// </summary>
-public class AzureCognitiveSearchMemory : ISemanticTextMemory
```
Comment (SergeyMenshykh) [inline dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemory.cs:25]: Just heads up: We stopped removing and renaming classes and their public members, if possible and if they are no longer needed and started marking them as obsolete instead to notify devs. using SK about upcoming changes and giving them time to adapt their code to use a new API.

**Comment by matthewbolanos** [comment]
@dluc, will this be resolved by #1174 ?

**Pull request merged**